diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 8ac05cb..0ce64bd 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -883,6 +883,8 @@ source_set("core") {
 
   if (is_ohos) {
     sources += [
+      "ui/ohos/shell_accessibility_ohos.cc",
+      "ui/ohos/shell_accessibility_ohos.h",
       "ui/ohos/aura_shell_runtime_bridge.cc",
       "ui/ohos/aura_shell_runtime_bridge.h",
       "ui/ohos/screen_orientation_delegate_ohos.cc",
@@ -1433,6 +1435,8 @@ source_set("core") {
     # chrome/browser/sync/sync_service_factory.cc), so link them in
     # explicitly here to avoid undefined-symbol errors.
     deps += [
+      "//components/ohos_accessibility",
+      "//ui/accessibility",
       "//chrome/browser/headless:headless_extra_parts",
       "//chrome/browser/security_events",
       "//chrome/browser/sensor",
