diff --git a/services/device/hid/BUILD.gn b/services/device/hid/BUILD.gn
--- a/services/device/hid/BUILD.gn
+++ b/services/device/hid/BUILD.gn
@@ -39,1 +39,12 @@ source_set("hid") {
+  if (is_ohos) {
+    sources += [
+      "hid_connection_ohos.cc",
+      "hid_connection_ohos.h",
+      "hid_service_ohos.cc",
+      "hid_service_ohos.h",
+    ]
+
+    deps += [ "//services/device/usb" ]
+  }
+
   if ((is_linux || is_chromeos) && use_udev) {
diff --git a/services/device/hid/hid_service.cc b/services/device/hid/hid_service.cc
--- a/services/device/hid/hid_service.cc
+++ b/services/device/hid/hid_service.cc
@@ -21,2 +21,4 @@
-#if (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && defined(USE_UDEV)
+#if BUILDFLAG(IS_OHOS)
+#include "services/device/hid/hid_service_ohos.h"
+#elif (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && defined(USE_UDEV)
 #include "services/device/hid/hid_service_linux.h"
@@ -71,2 +73,4 @@ std::unique_ptr<HidService> HidService::Create() {
-#if (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && defined(USE_UDEV)
+#if BUILDFLAG(IS_OHOS)
+  return std::make_unique<HidServiceOhos>();
+#elif (BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && defined(USE_UDEV)
   return std::make_unique<HidServiceLinux>();
@@ -87,4 +91,7 @@ std::unique_ptr<HidService> HidService::Create() {
 // static
 bool HidService::IsSupported() {
+#if BUILDFLAG(IS_OHOS)
+  return true;
+#endif
 #if ((BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)) && defined(USE_UDEV)) ||     BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN) || BUILDFLAG(IS_FUCHSIA) ||           BUILDFLAG(IS_ANDROID)
   return true;
diff --git a/services/device/usb/BUILD.gn b/services/device/usb/BUILD.gn
--- a/services/device/usb/BUILD.gn
+++ b/services/device/usb/BUILD.gn
@@ -18,3 +18,4 @@ static_library("usb") {
     ":*",
     "//services/device:*",
+    "//services/device/hid:*",
     "//services/device/usb/mojo:*",
