diff --git a/BUILD.gn b/BUILD.gn
index 3e1af3b..579a76b 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -729,7 +729,7 @@ angle_static_library("angle_gpu_info_util") {
     libs += [ "dxgi.lib" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_ohos) {
     sources += libangle_gpu_info_util_linux_sources
 
     if (angle_use_x11 && angle_has_build) {
diff --git a/gni/angle.gni b/gni/angle.gni
index 43ef5a9..91584a9 100644
--- a/gni/angle.gni
+++ b/gni/angle.gni
@@ -97,7 +97,7 @@ if (angle_has_build) {
     angle_use_gbm = ozone_platform_drm
     angle_use_x11 = ozone_platform_x11 && (is_linux || is_chromeos)
     angle_use_wayland = ozone_platform_wayland && is_linux && !is_castos
-    angle_use_vulkan_display = is_linux || is_chromeos
+    angle_use_vulkan_display = is_linux || is_chromeos || is_ohos
   }
 } else {
   declare_args() {
@@ -120,7 +120,7 @@ if (angle_has_build) {
   }
 
   declare_args() {
-    angle_use_vulkan_display = is_linux || is_chromeos
+    angle_use_vulkan_display = is_linux || is_chromeos || is_ohos
   }
 }
 
diff --git a/src/libANGLE/renderer/gl/BUILD.gn b/src/libANGLE/renderer/gl/BUILD.gn
index 6be9490..b2869f8 100644
--- a/src/libANGLE/renderer/gl/BUILD.gn
+++ b/src/libANGLE/renderer/gl/BUILD.gn
@@ -61,7 +61,7 @@ angle_source_set("angle_gl_backend") {
       "Xext",
     ]
   }
-  if (is_android || is_linux || is_chromeos) {
+  if (is_android || is_linux || is_chromeos || is_ohos) {
     deps += [ "$angle_root/src/common/linux:angle_dma_buf" ]
   }
   if (is_apple) {
diff --git a/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp b/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp
index 8ebc52f..206860e 100644
--- a/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp
+++ b/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp
@@ -118,6 +118,9 @@ const char *DisplayEGL::getEGLPath() const
 #    else
     return "/system/lib/libEGL.so";
 #    endif
+#elif defined(OS_OHOS) || defined(__OHOS__)
+    // HarmonyOS exposes the system EGL to apps as an NDK library.
+    return "libEGL.so";
 #else
     return "libEGL.so.1";
 #endif
diff --git a/src/libANGLE/renderer/gl/gl_backend.gni b/src/libANGLE/renderer/gl/gl_backend.gni
index 1d89bf7..09a4644 100644
--- a/src/libANGLE/renderer/gl/gl_backend.gni
+++ b/src/libANGLE/renderer/gl/gl_backend.gni
@@ -118,7 +118,7 @@ if (angle_use_x11) {
   ]
 }
 
-if (is_android || is_linux || is_chromeos) {
+if (is_android || is_linux || is_chromeos || is_ohos) {
   gl_backend_sources += [
     "egl/ContextEGL.cpp",
     "egl/ContextEGL.h",
diff --git a/src/libANGLE/renderer/vulkan/vulkan_backend.gni b/src/libANGLE/renderer/vulkan/vulkan_backend.gni
index 7ba7756..718547f 100644
--- a/src/libANGLE/renderer/vulkan/vulkan_backend.gni
+++ b/src/libANGLE/renderer/vulkan/vulkan_backend.gni
@@ -164,7 +164,7 @@ vulkan_backend_dma_buf_sources = [
   "linux/headless/WindowSurfaceVkHeadless.h",
 ]
 
-if (is_linux || is_chromeos) {
+if (is_linux || is_chromeos || is_ohos) {
   vulkan_backend_sources += vulkan_backend_dma_buf_sources
 }
 
diff --git a/src/libGLESv2.gni b/src/libGLESv2.gni
index ce5ab0b..4945028 100644
--- a/src/libGLESv2.gni
+++ b/src/libGLESv2.gni
@@ -116,7 +116,7 @@ if (is_android && angle_enable_unwind_backtrace_support) {
   libangle_common_sources += [ "src/common/backtrace_utils_noop.cpp" ]
 }
 
-if (is_linux || is_chromeos || is_android || is_fuchsia) {
+if (is_linux || is_chromeos || is_android || is_fuchsia || is_ohos) {
   libangle_common_sources += [
     "src/common/system_utils_linux.cpp",
     "src/common/system_utils_posix.cpp",
