diff --git a/src/base/base_paths.h b/src/base/base_paths.h
index 6742e3779588f..341db38370604 100644
--- a/src/base/base_paths.h
+++ b/src/base/base_paths.h
@@ -92,6 +92,7 @@ enum BasePathKey {
   DIR_TEST_DATA,           // Directory containing test data for //base tests.
                            // Only for use in base_unittests. Equivalent to
                            // DIR_SRC_TEST_DATA_ROOT + "/base/test/data".
+  DIR_SOURCE_ROOT = DIR_OUT_TEST_DATA_ROOT,
 
   PATH_END
 };
diff --git a/src/base/base_paths_ohos.cc b/src/base/base_paths_ohos.cc
index c8d58d17bebd5..91409686d13cf 100644
--- a/src/base/base_paths_ohos.cc
+++ b/src/base/base_paths_ohos.cc
@@ -6,6 +6,9 @@
 // OHOS in base/path_service.cc.
 #include "base/base_paths_ohos.h"
 
+#include <limits.h>
+#include <unistd.h>
+
 #include "base/base_paths.h"
 #include "base/base_switches.h"
 #include "base/command_line.h"
@@ -19,9 +22,12 @@
 namespace base {
 // TODO: temporary solution to load resources for multiple processes
 bool ParseAssetsOHOS(FilePath* result) {
-  auto bundle_path =
-      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
-          switches::kBundleInstallationDir);
+  if (!base::CommandLine::ForCurrentProcess()) {
+    LOG(ERROR) << "CommandLine not init";
+    return false;
+  }
+  auto bundle_path = base::CommandLine::ForCurrentProcess()->
+    GetSwitchValueASCII(switches::kBundleInstallationDir);
   if (bundle_path.empty()) {
     FilePath bin_dir;
     if (!ReadSymbolicLink(FilePath(kProcSelfExe), &bin_dir)) {
@@ -31,7 +37,12 @@ bool ParseAssetsOHOS(FilePath* result) {
     *result = bin_dir.DirName();
     return true;
   }
-  *result = FilePath(bundle_path.c_str());
+  bool for_test = base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kForTest);
+  if (for_test) {
+    *result = FilePath(bundle_path + "/entry/resources/rawfile");
+  } else {
+    *result = FilePath(bundle_path + "/nweb/entry/resources/rawfile");
+  }
   return true;
 }
 
@@ -46,27 +57,43 @@ bool PathProviderOHOS(int key, FilePath* result) {
       *result = bin_dir;
       return true;
     }
+    case base::FILE_MODULE:
+      NOTIMPLEMENTED();
+      return false;
     case base::DIR_MODULE: {
-      *result = base::FilePath();
+      if (!ReadSymbolicLink(FilePath(kProcSelfExe), &bin_dir)) {
+        NOTREACHED() << "Unable to resolve " << kProcSelfExe << ".";
+        return false;
+      }
+      *result = bin_dir.DirName();
       return true;
     }
+    case base::DIR_SOURCE_ROOT:
+      if (!ReadSymbolicLink(FilePath(kProcSelfExe), &bin_dir)) {
+        NOTREACHED() << "Unable to resolve " << kProcSelfExe << ".";
+        return false;
+      }
+      *result = bin_dir.DirName();
+      return true;
     case base::DIR_USER_DESKTOP:
-      *result =
-          FilePath(::ohos::adapter::ContextPathAdapter::GetUserDesktopDir());
-      return !result->empty();
+      NOTIMPLEMENTED();
+      return false;
     case base::DIR_CACHE:
-      *result = FilePath(::ohos::adapter::ContextPathAdapter::GetCacheDir());
-      return !result->empty();
-    case base::DIR_TEMP:
-      *result = FilePath(::ohos::adapter::ContextPathAdapter::GetTempDir());
-      return !result->empty();
+    #if BUILDFLAG(IS_ARKWEB) && BUILDFLAG(ARKWEB_COOKIE)
+      *result = FilePath("/data/storage/el2/base/cache/web");
+    #else
+      // set to /data/local directory for W|X permission.
+      *result = FilePath("/data/local");
+    #endif // #ifdef ARKWEB_COOKIE
+      return true;
     case base::DIR_ASSETS:
-      // TODO: temporary solution to load resources for multiple processes
       // resource file packed to system images
       return ParseAssetsOHOS(result);
     case base::DIR_OHOS_APP_DATA:
-      *result = FilePath(::ohos::adapter::ContextPathAdapter::GetFilesDir());
-      return !result->empty();
+      *result = FilePath("/data/local");
+      return true;
+    case base::DIR_OHOS_EXTERNAL_STORAGE:
+      return false;
     default:
       return false;
   }
diff --git a/src/base/base_paths_ohos.h b/src/base/base_paths_ohos.h
index cf363018c0d9a..d72fcae59da0b 100644
--- a/src/base/base_paths_ohos.h
+++ b/src/base/base_paths_ohos.h
@@ -5,11 +5,20 @@
 #ifndef BASE_BASE_PATHS_OHOS_H_
 #define BASE_BASE_PATHS_OHOS_H_
 
+#include "arkweb/build/features/features.h"
+#include "build/build_config.h"
+
 namespace base {
 
 enum {
   PATH_OHOS_START = 500,
+
   DIR_OHOS_APP_DATA,
+  DIR_OHOS_EXTERNAL_STORAGE,
+  DIR_OHOS_APP_INSTALLATION,
+#if BUILDFLAG(IS_ARKWEB) && BUILDFLAG(ARKWEB_CRASHPAD)
+  DIR_OHOS_CRASHPAD,
+#endif
   PATH_OHOS_END
 };
 
