diff --git a/src/base/BUILD.gn b/src/base/BUILD.gn
index 5cefe3865d5b9..050cb68afbef3 100644
--- a/src/base/BUILD.gn
+++ b/src/base/BUILD.gn
@@ -62,6 +62,12 @@ if (is_win) {
   import("//build/config/win/control_flow_guard.gni")
 }
 
+import("//arkweb/build/features/features.gni")
+
+if (enable_arkweb) {
+  import("//arkweb/chromium_ext/base/arkweb_base_ext.gni")
+}
+
 declare_args() {
   # Unsafe developer build. Has developer-friendly features that may weaken or
   # disable security measures like sandboxing or ASLR.
@@ -1068,6 +1074,10 @@ component("base") {
     deps += [ ":rust_logger" ]
   }
 
+  if (enable_arkweb) {
+    deps += [ "//arkweb//build/features" ]
+  }
+
   # `raw_ptr` cannot be made a component due to CRT symbol issues.
   # Its gateway to being a component is through `//base`, so we have
   # to provide the appropriate `#define` here.
@@ -1107,6 +1117,20 @@ component("base") {
     "//build:chromeos_buildflags",
     "//third_party/abseil-cpp:absl",
   ]
+  if (is_ohos && enable_arkweb) {
+    sources += arkweb_base_ext_sources
+    deps += [ "//arkweb//build/features" ]
+    public_deps += [ "//arkweb//build/features" ]
+    public_deps += [ ":ohos_package_info" ]
+    if (is_ohos) {
+      if (!defined(public_deps)) {
+        public_deps = []
+      }
+      public_deps += [
+        "//ohos_glue:ohos_adapter_glue_source",
+      ]
+    }
+  }
 
   if (build_rust_base_conversions) {
     sources += [
@@ -2379,7 +2403,7 @@ component("base") {
     sources += [ "time/time_exploded_posix.cc" ]
   }
 
-  if ((is_posix && !is_apple && !is_android) || is_fuchsia) {
+  if ((is_posix && !is_apple && !is_android && !enable_arkweb) || is_fuchsia) {
     sources += [ "profiler/stack_sampler_posix.cc" ]
   }
 
@@ -2587,6 +2611,15 @@ if (is_linux || is_chromeos) {
   }
 }
 
+# Build flags for ArkWeb package infomation
+if (enable_arkweb) {
+  buildflag_header("ohos_package_info") {
+    header = "ohos_package_info.h"
+    header_dir = "base/ohos"
+    flags = [ "OHOS_PACKAGE_NAME=\"${ohos_package_name}\"" ]
+  }
+}
+
 # Build flags for Control Flow Integrity
 # https://www.chromium.org/developers/testing/control-flow-integrity
 buildflag_header("cfi_buildflags") {
diff --git a/src/base/message_loop/message_pump_for_ui.h b/src/base/message_loop/message_pump_for_ui.h
index 09d95d592675a..64b251ddeda27 100644
--- a/src/base/message_loop/message_pump_for_ui.h
+++ b/src/base/message_loop/message_pump_for_ui.h
@@ -14,6 +14,8 @@
 #include "base/message_loop/message_pump_win.h"
 #elif BUILDFLAG(IS_ANDROID)
 #include "base/message_loop/message_pump_android.h"
+#elif BUILDFLAG(IS_ARKWEB)
+#include "base/message_loop/message_pump_ohos.h"
 #elif BUILDFLAG(IS_APPLE)
 #include "base/message_loop/message_pump.h"
 #elif BUILDFLAG(IS_NACL) || BUILDFLAG(IS_AIX)
@@ -34,6 +36,8 @@ namespace base {
 using MessagePumpForUI = MessagePumpForUI;
 #elif BUILDFLAG(IS_ANDROID)
 using MessagePumpForUI = MessagePumpAndroid;
+#elif BUILDFLAG(IS_ARKWEB)
+using MessagePumpForUI = MessagePumpForUI;
 #elif BUILDFLAG(IS_APPLE)
 // MessagePumpForUI isn't bound to a specific impl on Mac. While each impl can
 // be represented by a plain MessagePump: message_pump_apple::Create() must be
diff --git a/src/base/run_loop.h b/src/base/run_loop.h
index a977c3722f4bf..728b5543bd2f1 100644
--- a/src/base/run_loop.h
+++ b/src/base/run_loop.h
@@ -34,6 +34,10 @@ class ScopedDisableRunLoopTimeout;
 class MessagePumpAndroid;
 #endif
 
+#if BUILDFLAG(IS_ARKWEB)
+class MessagePumpForUI;
+#endif
+
 #if BUILDFLAG(IS_IOS)
 class MessagePumpUIApplication;
 #endif
@@ -271,6 +275,10 @@ class BASE_EXPORT RunLoop {
   friend class MessagePumpAndroid;
 #endif
 
+#if BUILDFLAG(IS_ARKWEB)
+  friend class MessagePumpForUI;
+#endif
+
 #if BUILDFLAG(IS_IOS)
   // iOS doesn't support the blocking RunLoop::Run, so it calls
   // BeforeRun directly.
diff --git a/src/base/task/current_thread.cc b/src/base/task/current_thread.cc
index c8c80509abc91..2042451c31bfa 100644
--- a/src/base/task/current_thread.cc
+++ b/src/base/task/current_thread.cc
@@ -154,7 +154,8 @@ MessagePumpForUI* CurrentUIThread::GetMessagePumpForUI() const {
   return static_cast<MessagePumpForUI*>(current_->GetMessagePump());
 }
 
-#if BUILDFLAG(IS_OZONE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_OZONE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_WIN) && \
+    !BUILDFLAG(IS_ARKWEB)
 bool CurrentUIThread::WatchFileDescriptor(
     int fd,
     bool persistent,
diff --git a/src/base/task/current_thread.h b/src/base/task/current_thread.h
index cdadebb201547..58c0b4b9dc5f2 100644
--- a/src/base/task/current_thread.h
+++ b/src/base/task/current_thread.h
@@ -260,7 +260,8 @@ class BASE_EXPORT CurrentUIThread : public CurrentThread {
 
   CurrentUIThread* operator->() { return this; }
 
-#if BUILDFLAG(IS_OZONE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_OZONE) && !BUILDFLAG(IS_FUCHSIA) && !BUILDFLAG(IS_WIN) && \
+    !BUILDFLAG(IS_ARKWEB)
   static_assert(
       std::is_base_of_v<WatchableIOMessagePumpPosix, MessagePumpForUI>,
       "CurrentThreadForUI::WatchFileDescriptor is supported only"
diff --git a/src/base/task/sequence_manager/sequence_manager_impl.cc b/src/base/task/sequence_manager/sequence_manager_impl.cc
index 6de397fef68b1..db21a0d07acca 100644
--- a/src/base/task/sequence_manager/sequence_manager_impl.cc
+++ b/src/base/task/sequence_manager/sequence_manager_impl.cc
@@ -321,6 +321,10 @@ void SequenceManagerImpl::BindToMessagePump(std::unique_ptr<MessagePump> pump) {
       settings_.message_loop_type == MessagePumpType::JAVA) {
     controller_->AttachToMessagePump();
   }
+#elif BUILDFLAG(IS_ARKWEB)
+  if (settings_.message_loop_type == MessagePumpType::UI) {
+    controller_->AttachToMessagePump();
+  }
 #endif
 
   // On iOS attach to the native loop when there is one.
diff --git a/src/base/task/sequence_manager/thread_controller.h b/src/base/task/sequence_manager/thread_controller.h
index c7f95f60a1d63..1b7ca33a26cd6 100644
--- a/src/base/task/sequence_manager/thread_controller.h
+++ b/src/base/task/sequence_manager/thread_controller.h
@@ -132,7 +132,7 @@ class BASE_EXPORT ThreadController {
   // Returns true if the current run loop should quit when idle.
   virtual bool ShouldQuitRunLoopWhenIdle() = 0;
 
-#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_ARKWEB)
   // On iOS, the main message loop cannot be Run().  Instead call
   // AttachToMessagePump(), which connects this ThreadController to the
   // UI thread's CFRunLoop and allows PostTask() to work.
diff --git a/src/base/task/sequence_manager/thread_controller_impl.cc b/src/base/task/sequence_manager/thread_controller_impl.cc
index 54ca4531280c7..eed5ccc85ad33 100644
--- a/src/base/task/sequence_manager/thread_controller_impl.cc
+++ b/src/base/task/sequence_manager/thread_controller_impl.cc
@@ -355,7 +355,7 @@ MessagePump* ThreadControllerImpl::GetBoundMessagePump() const {
   return nullptr;
 }
 
-#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_ARKWEB)
 void ThreadControllerImpl::AttachToMessagePump() {
   NOTREACHED();
 }
diff --git a/src/base/task/sequence_manager/thread_controller_impl.h b/src/base/task/sequence_manager/thread_controller_impl.h
index 0de16de103705..54d98dabaead1 100644
--- a/src/base/task/sequence_manager/thread_controller_impl.h
+++ b/src/base/task/sequence_manager/thread_controller_impl.h
@@ -60,7 +60,7 @@ class BASE_EXPORT ThreadControllerImpl : public ThreadController,
   void SetTaskExecutionAllowedInNativeNestedLoop(bool allowed) override;
   bool IsTaskExecutionAllowed() const override;
   MessagePump* GetBoundMessagePump() const override;
-#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_ARKWEB)
   void AttachToMessagePump() override;
 #endif
 #if BUILDFLAG(IS_IOS)
diff --git a/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.cc b/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.cc
index e9fe13844af65..d11086911f6d2 100644
--- a/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.cc
+++ b/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.cc
@@ -735,7 +735,7 @@ void ThreadControllerWithMessagePumpImpl::AttachToMessagePump() {
 void ThreadControllerWithMessagePumpImpl::DetachFromMessagePump() {
   static_cast<MessagePumpCFRunLoopBase*>(pump_.get())->Detach();
 }
-#elif BUILDFLAG(IS_ANDROID)
+#elif BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_ARKWEB)
 void ThreadControllerWithMessagePumpImpl::AttachToMessagePump() {
   CHECK(main_thread_only().work_batch_size == 1);
   // Aborting the message pump currently relies on the batch size being 1.
diff --git a/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.h b/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.h
index ae8f67ed83a1d..8dcb4793fa628 100644
--- a/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.h
+++ b/src/base/task/sequence_manager/thread_controller_with_message_pump_impl.h
@@ -73,7 +73,7 @@ class BASE_EXPORT ThreadControllerWithMessagePumpImpl
   bool IsTaskExecutionAllowed() const override;
   MessagePump* GetBoundMessagePump() const override;
   void PrioritizeYieldingToNative(base::TimeTicks prioritize_until) override;
-#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_IOS) || BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_ARKWEB)
   void AttachToMessagePump() override;
 #endif
 #if BUILDFLAG(IS_IOS)
