diff --git a/src/base/threading/platform_thread.h b/src/base/threading/platform_thread.h
index bb0b94e2f2f69..19ba610a19be4 100644
--- a/src/base/threading/platform_thread.h
+++ b/src/base/threading/platform_thread.h
@@ -166,6 +166,11 @@ class BASE_EXPORT PlatformThreadBase {
   // Gets the current thread id, which may be useful for logging purposes.
   static PlatformThreadId CurrentId();
 
+#if BUILDFLAG(IS_ARKWEB)
+  // Gets the current thread global id, which may be useful for logging purposes.
+  static PlatformThreadId CurrentRealId();
+#endif
+
   // Gets the current thread reference, which can be used to check if
   // we're on the right thread quickly.
   static PlatformThreadRef CurrentRef();
diff --git a/src/base/threading/platform_thread_posix.cc b/src/base/threading/platform_thread_posix.cc
index 33e88fcaefa42..cfc7b3e8e9d0d 100644
--- a/src/base/threading/platform_thread_posix.cc
+++ b/src/base/threading/platform_thread_posix.cc
@@ -12,6 +12,8 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <string.h>
+#include <dlfcn.h>
 
 #include <memory>
 #include <tuple>
@@ -270,6 +272,20 @@ PlatformThreadId PlatformThreadBase::CurrentId() {
 #endif
 }
 
+#if BUILDFLAG(IS_ARKWEB)
+NO_SANITIZE("cfi-icall") PlatformThreadId PlatformThread::CurrentRealId() {
+  // - getproctid() is fast, since its return value is cached in pthread (in the
+  //   thread control block of pthread). See gettid.c in bionic.
+  using GetProcXid = int (*)(void);
+  static GetProcXid getProcTid = nullptr;
+  if (getProcTid == nullptr) {
+    getProcTid = reinterpret_cast<GetProcXid>(dlsym(RTLD_DEFAULT, "getproctid"));
+    CHECK(getProcTid);
+  }
+  return getProcTid();
+}
+#endif
+
 // static
 PlatformThreadRef PlatformThreadBase::CurrentRef() {
   return PlatformThreadRef(pthread_self());
diff --git a/src/base/threading/simple_thread.cc b/src/base/threading/simple_thread.cc
index d4954df21cd1a..5f2e9b3e07839 100644
--- a/src/base/threading/simple_thread.cc
+++ b/src/base/threading/simple_thread.cc
@@ -64,12 +64,22 @@ PlatformThreadId SimpleThread::tid() {
   return tid_;
 }
 
+#if BUILDFLAG(IS_ARKWEB)
+PlatformThreadId SimpleThread::RealTid() {
+  DCHECK(HasBeenStarted());
+  return realTid_;
+}
+#endif
+
 bool SimpleThread::HasBeenStarted() {
   return event_.IsSignaled();
 }
 
 void SimpleThread::ThreadMain() {
   tid_ = PlatformThread::CurrentId();
+#if BUILDFLAG(IS_ARKWEB)
+  realTid_ = PlatformThread::CurrentRealId();
+#endif
   PlatformThread::SetName(name_);
 
   // We've initialized our new thread, signal that we're done to Start().
diff --git a/src/base/threading/simple_thread.h b/src/base/threading/simple_thread.h
index 6c23811fd3b0c..842291e1bf129 100644
--- a/src/base/threading/simple_thread.h
+++ b/src/base/threading/simple_thread.h
@@ -118,6 +118,14 @@ class BASE_EXPORT SimpleThread : public PlatformThread::Delegate {
   // be called before HasBeenStarted() returns True.
   PlatformThreadId tid();
 
+#if BUILDFLAG(IS_ARKWEB)
+  // Returns the thread global id, only valid after the thread has started. If the
+  // thread was started using Start(), then this will be valid after the call to
+  // Start(). If StartAsync() was used to start the thread, then this must not
+  // be called before HasBeenStarted() returns True.
+  PlatformThreadId RealTid();
+#endif
+
   // Returns True if the thread has been started and initialized (i.e. if
   // ThreadMain() has run). If the thread was started with StartAsync(), but it
   // hasn't been initialized yet (i.e. ThreadMain() has not run), then this will
@@ -151,6 +159,9 @@ class BASE_EXPORT SimpleThread : public PlatformThread::Delegate {
   PlatformThreadHandle thread_;  // PlatformThread handle, reset after Join.
   WaitableEvent event_;          // Signaled if Start() was ever called.
   PlatformThreadId tid_ = kInvalidThreadId;  // The backing thread's id.
+#if BUILDFLAG(IS_ARKWEB)
+  PlatformThreadId realTid_ = kInvalidThreadId;  // The backing thread's global id.
+#endif
   bool joined_ = false;                      // True if Join has been called.
   // Set to true when the platform-thread creation has started.
   bool start_called_ = false;
diff --git a/src/base/threading/thread.cc b/src/base/threading/thread.cc
index b75ef9e4477d8..352a76f8b3253 100644
--- a/src/base/threading/thread.cc
+++ b/src/base/threading/thread.cc
@@ -183,7 +183,9 @@ bool Thread::StartWithOptions(Options options) {
   // Reset |id_| here to support restarting the thread.
   id_event_.Reset();
   id_ = kInvalidThreadId;
-
+#if BUILDFLAG(IS_ARKWEB)
+  real_id_ = kInvalidThreadId;
+#endif
   SetThreadWasQuitProperly(false);
 
   if (options.delegate) {
@@ -312,6 +314,17 @@ PlatformThreadId Thread::GetThreadId() const {
   return id_;
 }
 
+#if BUILDFLAG(IS_ARKWEB)
+PlatformThreadId Thread::GetThreadRealId() const {
+  if (!id_event_.IsSignaled()) {
+    // If the thread is created but not started yet, wait for |id_| being ready.
+    base::ScopedAllowBaseSyncPrimitivesOutsideBlockingScope allow_wait;
+    id_event_.Wait();
+  }
+  return real_id_;
+}
+#endif
+
 bool Thread::IsRunning() const {
   // TODO(gab): Fix improper usage of this API (http://crbug.com/629139) and
   // enable this check.
@@ -362,6 +375,9 @@ void Thread::ThreadMain() {
   // write in StartWithOptions().
   DCHECK_EQ(kInvalidThreadId, id_);
   id_ = PlatformThread::CurrentId();
+#if BUILDFLAG(IS_ARKWEB)
+  real_id_ = PlatformThread::CurrentRealId();
+#endif
   DCHECK_NE(kInvalidThreadId, id_);
   id_event_.Signal();
 
diff --git a/src/base/threading/thread.h b/src/base/threading/thread.h
index 2b4fae7d22afd..13fc990a1ddaa 100644
--- a/src/base/threading/thread.h
+++ b/src/base/threading/thread.h
@@ -253,6 +253,11 @@ class BASE_EXPORT Thread : PlatformThread::Delegate {
   // This method is thread-safe.
   PlatformThreadId GetThreadId() const;
 
+#if BUILDFLAG(IS_ARKWEB)
+  // This method is thread-safe.
+  PlatformThreadId GetThreadRealId() const;
+#endif
+
   // Returns true if the thread has been started, and not yet stopped.
   bool IsRunning() const;
 
@@ -313,6 +318,12 @@ class BASE_EXPORT Thread : PlatformThread::Delegate {
 
   // The thread's id once it has started.
   PlatformThreadId id_ = kInvalidThreadId;
+
+#if BUILDFLAG(IS_ARKWEB)
+  // The thread's global id once it has started.
+  PlatformThreadId real_id_ = kInvalidThreadId;
+#endif
+
   // Protects |id_| which must only be read while it's signaled.
   mutable WaitableEvent id_event_;
 
