diff --git a/BUILD.gn b/BUILD.gn
index 357a4d154..62f91af56 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -120,6 +120,22 @@ group("gn_all") {
       "//url:url_unittests",
     ]
 
+    #ifdef OHOS_NWEB_EX
+    if (defined(ohos_nweb_ex) && ohos_nweb_ex) {
+      deps += [
+        "//arkweb/ohos_browser_shell",
+        "//arkweb/ohos_nweb_ex/browser_service",
+        "//arkweb/ohos_nweb_ex/test:ohos_nweb_ex_unittests",
+        "//third_party/ohos_nweb_hap",
+      ]
+    deps += [
+     # "//ohos_test",
+    ]
+    }  # endif // OHOS_NWEB_EX
+
+    if (defined(enable_ohos_nweb_hap) && enable_ohos_nweb_hap) {
+      deps += [ "//third_party/ohos_nweb_hap" ]
+    }
     if (!is_component_build) {
       deps += [ "//third_party/abseil-cpp:absl_tests" ]
     }
diff --git a/build/config/BUILD.gn b/build/config/BUILD.gn
index 3a72c528d..f7ff2ea38 100644
--- a/build/config/BUILD.gn
+++ b/build/config/BUILD.gn
@@ -106,6 +106,25 @@ config("feature_flags") {
 
 # Debug/release ----------------------------------------------------------------
 
+#ifdef OHOS_NWEB_EX
+config("ohos_nweb_ex_def_main") {
+  if (defined(ohos_nweb_ex_config_name) && ohos_nweb_ex_config_name != "") {
+    defines = ohos_nweb_ex_defines_main
+  }
+}
+
+config("ohos_nweb_ex_def_blink") {
+  if (defined(ohos_nweb_ex_config_name) && ohos_nweb_ex_config_name != "") {
+    defines = ohos_nweb_ex_defines_blink_core
+  }
+}
+
+config("ohos_nweb_ex_def_other") {
+  if (defined(ohos_nweb_ex_config_name) && ohos_nweb_ex_config_name != "") {
+    defines = ohos_nweb_ex_defines_other
+  }
+}  #endif  // ohos_nweb_ex
+
 config("debug") {
   defines = [
     "_DEBUG",
diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index d4d3abd39..7ac4f916c 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -46,6 +46,15 @@
 # When writing build files, to do something only for the host:
 #   if (current_toolchain == host_toolchain) { ...
 
+#ifdef OHOS_NWEB_EX
+declare_args() {
+  ohos_nweb_ex_config_name = ""
+}
+
+if (ohos_nweb_ex_config_name != "") {
+  import("$ohos_nweb_ex_config_name")
+}  #endif  // OHOS_NWEB_EX
+
 if (target_os == "") {
   target_os = host_os
 }
@@ -412,6 +421,11 @@ if (is_debug) {
   default_compiler_configs += [ "//build/config:release" ]
 }
 
+#ifdef OHOS_NWEB_EX
+if (ohos_nweb_ex_config_name != "") {
+  default_compiler_configs += [ "//build/config:ohos_nweb_ex_def_main" ]
+} #endif //OHOS_NWEB_EX
+
 # Static libraries and source sets use only the compiler ones.
 set_defaults("static_library") {
   configs = default_compiler_configs
