diff --git a/chromium_src_1023/chromium_src/components/crash/core/app/BUILD.gn b/chromium_126.0.6478.36_dev_1021/src/components/crash/core/app/BUILD.gn
index 023f142..89c3aaa 100644
--- a/chromium_src_1023/chromium_src/components/crash/core/app/BUILD.gn
+++ b/chromium_126.0.6478.36_dev_1021/src/components/crash/core/app/BUILD.gn
@@ -82,6 +82,10 @@ static_library("app") {
     ]
   }
 
+  if (defined(ohos_crashpad) && ohos_crashpad) {
+    sources += [ "crashpad_ohos.cc" ]
+  }
+
   defines = [ "CRASH_IMPLEMENTATION" ]
 
   public_deps = [ ":lib" ]
@@ -100,10 +104,6 @@ static_library("app") {
     data_deps = [ ":chrome_crashpad_handler" ]
   }
 
-  if (is_ohos) {
-    sources += [ "crashpad_linux.cc" ]
-  }
-
   if (is_ios) {
     deps -= [ "//content/public/common:content_descriptors" ]
   }
@@ -281,6 +281,14 @@ if (is_mac || is_android || is_linux || is_chromeos) {
   }
 }
 
+if (defined(ohos_crashpad) && ohos_crashpad) {
+  shared_library("arkweb_crashpad_handler") {
+    deps = [
+      ":crashpad_handler_main",
+    ]
+  }
+}
+
 if (is_android) {
   # There is not any normal way to package native executables in an Android APK.
   # It is normal to package native code as a loadable module but Android's APK
