diff --git a/chromium_src_1023/chromium_src/components/crash/core/app/crash_reporter_client.cc b/chromium_126.0.6478.36_dev_1021/src/components/crash/core/app/crash_reporter_client.cc
index 05bbc81..6420d78 100644
--- a/chromium_src_1023/chromium_src/components/crash/core/app/crash_reporter_client.cc
+++ b/chromium_126.0.6478.36_dev_1021/src/components/crash/core/app/crash_reporter_client.cc
@@ -165,7 +165,7 @@ bool CrashReporterClient::EnableBrowserCrashForwarding() {
 }
 #endif
 
-#if BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_OHOS)
 unsigned int CrashReporterClient::GetCrashDumpPercentage() {
   return 100;
 }
@@ -202,7 +202,7 @@ bool CrashReporterClient::ShouldWriteMinidumpToLog() {
 
 #endif
 
-#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_OHOS)
 void CrashReporterClient::GetSanitizationInformation(
     const char* const** allowed_annotations,
     void** target_module,
